### PR TITLE
Land surface upgrades for HR4

### PR DESCRIPTION
**Pull request by HelinWei-NOAA**
To address some concerns from HR3 especially the surface cold biases we have finalized some land surface upgrades after carrying out extensive tests and evaluation.

1. remove vegetation type-dependent for the diagnostic option (opt_diag) used in HR3
2. use radiative temperature as skin temperature
3. use single loop alternative approach to address some flux inconsistence below and above the canopy
4. fix the thermal roughness bug (the one for bared portion in the composition is not the same as the one used in the bared title)
5. adjust some momentum roughness length
6. use the interpolated temp between the skin temp and the first soil temp to get the initial snow temp based on snow depth
7. use constant snow thermal conductivity to reduce the snow insulation effects
8. fix radiation parameter bugs for grassland/cropland
9. adjust albedo parameter for each soil color category

### CODE REVIEW

**Comment by yangfanglin** [comment]
@barlage @cenlinhe @dustinswales @grantfirl  We'd appreciate it if you can review this PR.  We'd like to get this PR committed to start GFS HR4 experiments at the earliest time possible.

File: physics/Interstitials/UFS_SCM_NEPTUNE/GFS_phys_time_vary.fv3.F90 [hunk 1]
```diff
@@ -593,8 +593,24 @@ subroutine GFS_phys_time_vary_init (
...
+!                  accumulated_thickness = accumulated_thickness + dzsno(is)
+!                  mid_points(is) = accumulated_thickness - dzsno(is) / 2.0
+!                  layer_temp = tgxy(ix) + (mid_points(is) / total_thickness) * temp_diff
+!                  tsnoxy(ix,is) = layer_temp
+!                end do
+
                  do is = isnow,0
-                   tsnoxy(ix,is)  = tgxy(ix)
+!                  tsnoxy(ix,is)  = tgxy(ix)
+!                  tsnoxy(ix,is) =  tgxy(ix) + (( sum(dzsno(isnow:is)) -0.5*dzsno(is) )/snd)*(tgxy(ix)-stc(ix,1))
```
Comment (barlage) [inline physics/Interstitials/UFS_SCM_NEPTUNE/GFS_phys_time_vary.fv3.F90]: probably best to clean this up by removing all the comments

**Review by barlage — APPROVED** [review]

File: physics/SFC_Models/Land/Noahmp/module_sf_noahmplsm.F90 [hunk 2]
```diff
@@ -1989,7 +1989,7 @@ subroutine energy (parameters,ice    ,vegtyp ,ist    ,nsnow  ,nsoil  , & !in
 
   real (kind=kind_phys), parameter                   :: mpe    = 1.e-6
   real (kind=kind_phys), parameter                   :: psiwlt = -150.  !metric potential for wilting point (m)
-  real (kind=kind_phys), parameter                   :: z0     = 0.002  ! bare-soil roughness length (m) (i.e., under the canopy)
+  real (kind=kind_phys), parameter                   :: z0     = 0.015  ! bare-soil roughness length (m) (i.e., under the canopy)
```
Comment (cenlinhe) [inline physics/SFC_Models/Land/Noahmp/module_sf_noahmplsm.F90:1992]: where does "0.015" come from? Is it tuned? I did not find this value in the Noah-MP.

Comment (barlage) [reply physics/SFC_Models/Land/Noahmp/module_sf_noahmplsm.F90:1992]: it is tuned based on wind speed bias over bare soil and vegetated areas with low vegetation fraction

**Review by cenlinhe — APPROVED** [review]
The changes look good to me.

File: physics/SFC_Models/Land/Noahmp/module_sf_noahmplsm.F90 (hunk 2 above)
Comment (cenlinhe) [reply physics/SFC_Models/Land/Noahmp/module_sf_noahmplsm.F90:1992]: got it. thanks!

File: physics/Interstitials/UFS_SCM_NEPTUNE/GFS_phys_time_vary.fv3.F90 (hunk 1 above)
Comment (HelinWei-NOAA) [reply physics/Interstitials/UFS_SCM_NEPTUNE/GFS_phys_time_vary.fv3.F90]: Done

**Comment by jkbk2004** [comment]
@grantfirl @dustinswales any chance for final reviews?

**Review by grantfirl — APPROVED** [review]

**Comment by FernandoAndrade-NOAA** [comment]
Testing for #[2354](https://github.com/ufs-community/ufs-weather-model/pull/2354) has completed successfully, please continue with the merge process. @grantfirl

**Pull request merged**